group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorResult.java

```diff
@@ -44,8 +44,8 @@ public class CoordinatorResult<T, U> {
     private final CompletableFuture<Void> appendFuture;
 
     /**
-     * The boolean indicating whether to replay the records. Without specifying.
-     * The default value is {@code appendFuture == null}.
+     * The boolean indicating whether to replay the records.
+     * The default value is {@code true} unless specified otherwise.
      */
     private final Boolean replayRecords;
```

Comment:
We should use a primitive type here.

Reply:
This was because we have to call `replayRecords.hashCode()` to compute the CoordinatorResult's hash.

Reply:
I think that you could use `result = 31 * result + (replayRecords ? 1 : 0);` instead.

Reply:
Should we use 
```
    public static int hashCode(boolean value) {
        return value ? 1231 : 1237;
    }
```

Reply:
I am not sure. The one that I gave was auto-generated by intellij.

Reply:
Then let's go with `replayRecords ? 1 : 0`. I guess it doesn't matter.